libcloudforensics/providers/gcp/internal/storage.py

```diff
@@ -173,3 +175,63 @@ def DeleteObject(self, gcs_path: str) -> None:
     gcs_objects = self.GcsApi().objects()
     request = gcs_objects.delete(bucket=bucket, object=object_path)
     request.execute()  # type: Dict[str, Any]
+
+  def GetBucketSize(self,
+                    bucket_name: str = None,
+                    timeframe: int = 1) -> Dict[str, int]:
+    """List the size of a Google Storage Bucket in a project (default: last 1 day).
+
+    Note: This will list the maximum size the bucket had in the timeframe.
+
+    Ref: https://cloud.google.com/monitoring/api/metrics_gcp#gcp-storage
+
+    Args:
+      bucket_name (str):  Name of a bucket in GCS.
+      timeframe (int): Optional. The number (in days) for
+          which to measure activity.
+
+    Returns:
+      Dict[str, int]: Dictionary mapping bucket name to its size.
+    """
+
+    start_time = common.FormatRFC3339(
+        datetime.datetime.utcnow() - datetime.timedelta(days=timeframe))
+    end_time = common.FormatRFC3339(datetime.datetime.utcnow())
+    period = timeframe * 24 * 60 * 60
+
+    gcm = gcp_monitoring.GoogleCloudMonitoring(self.project_id)
+    gcm_api = gcm.GcmApi()
+    gcm_timeseries_client = gcm_api.projects().timeSeries()
+    qfilter = 'metric.type="storage.googleapis.com/storage/total_bytes" resource.type="gcs_bucket"'
+
+    if bucket_name:
+      qfilter += ' resource.label.bucket_name="{0:s}"'.format(bucket_name)
+
+    responses = common.ExecuteRequest(
+        gcm_timeseries_client,
+        'list',
+        {
+            'name': 'projects/{0:s}'.format(self.project_id),
+            'filter': qfilter,
+            'interval_startTime': start_time,
+            'interval_endTime': end_time,
+            'aggregation_groupByFields': 'resource.label.bucket_name',
+            'aggregation_perSeriesAligner': 'ALIGN_MAX',
+            'aggregation_alignmentPeriod': '{0:d}s'.format(period),
+            'aggregation_crossSeriesReducer': 'REDUCE_NONE'
+        })
+
+    ret = {}
+    for response in responses:
+      for ts in response.get('timeSeries', []):
+        bucket = ts.get('resource', {}).get('labels', {}).get('bucket_name', '')
+        if bucket:
+          points = ts.get('points', [])
+          if points:
```

Comment:
Is the `if points` statement necessary here? If `points = ts.get('points', [])` returns an empty list, then the following `for` loop will be ignored anyways

Reply:
Done